File: aidlc-table-order/backend/app/dto/order.py
"""Order DTOs — request/response models for C5: Order."""

from datetime import datetime

from pydantic import BaseModel


class OrderItemOptionRequest(BaseModel):
    option_item_id: str


class OrderItemRequest(BaseModel):
    menu_id: str
    quantity: int
    options: list[OrderItemOptionRequest] = []


class OrderCreateRequest(BaseModel):
    items: list[OrderItemRequest]


class OrderItemOptionResponse(BaseModel):
    option_group_name: str
    option_item_name: str
    extra_price: int


class OrderItemResponse(BaseModel):
    id: str
    menu_name: str
    quantity: int
    unit_price: int
    option_total_price: int
    subtotal: int
    options: list[OrderItemOptionResponse]


class OrderResponse(BaseModel):
    id: str
    order_number: str
    round: int
    status: str
    total_amount: int
    created_at: datetime
    items: list[OrderItemResponse]


class OrderStatusUpdateRequest(BaseModel):
    status: str


class PollingOrderPreview(BaseModel):
    menu_name: str
    quantity: int


class PollingTableInfo(BaseModel):
    table_id: str
    table_number: int
    total_amount: int
    recent_orders: list[PollingOrderPreview]
    has_new: bool


class PollingResponse(BaseModel):
    tables: list[PollingTableInfo]
    polled_at: datetime


class ReceiptRound(BaseModel):
    round: int
    orders: list[OrderResponse]


class ReceiptResponse(BaseModel):
    rounds: list[ReceiptRound]
    grand_total: int

File: aidlc-table-order/packages/shared/src/types/order.ts
import type { OrderStatus } from './common';

export interface OrderItemOptionRequest {
  option_item_id: string;
}

export interface OrderItemRequest {
  menu_id: string;
  quantity: number;
  options: OrderItemOptionRequest[];
}

export interface CreateOrderRequest {
  items: OrderItemRequest[];
}

export interface Order {
  id: string;
  store_id: string;
  table_id: string;
  session_id: string;
  order_number: string;
  round: number;
  status: OrderStatus;
  total_amount: number;
  created_at: string;
  updated_at: string;
}

export interface OrderItemOption {
  id: string;
  option_group_name: string;
  option_item_name: string;
  extra_price: number;
}

export interface OrderItem {
  id: string;
  order_id: string;
  menu_id: string;
  menu_name: string;
  quantity: number;
  unit_price: number;
  option_total_price: number;
  subtotal: number;
  options: OrderItemOption[];
}

export interface OrderWithItems extends Order {
  items: OrderItem[];
}

export interface SessionOrdersResponse {
  orders: OrderWithItems[];
  total_amount: number;
}

export interface UpdateOrderStatusRequest {
  status: OrderStatus;
}

export interface ReceiptResponse {
  session_id: string;
  table_number: number;
  rounds: ReceiptRound[];
  total_amount: number;
}

export interface ReceiptRound {
  round: number;
  order_number: string;
  ordered_at: string;
  items: OrderItem[];
  round_total: number;
}

export interface PollingTableData {
  table_id: string;
  table_number: number;
  session_id: string | null;
  total_amount: number;
  recent_orders: PollingOrderPreview[];
  has_new: boolean;
}

export interface PollingOrderPreview {
  order_id: string;
  order_number: string;
  round: number;
  status: OrderStatus;
  summary: string;
  total_amount: number;
  created_at: string;
}

export interface PollingResponse {
  tables: PollingTableData[];
  timestamp: string;
}

export interface OrderProgressResponse {
  order_id: string;
  status: OrderStatus;
  estimated_total_seconds: number;
  elapsed_seconds: number;
  progress_percent: number;
}

export interface OrderHistory {
  id: string;
  original_order_id: string;
  store_id: string;
  table_id: string;
  session_id: string;
  order_number: string;
  round: number;
  total_amount: number;
  items_snapshot: OrderItem[];
  ordered_at: string;
  archived_at: string;
}
